src/sitemap/handler.js

```diff
@@ -10,131 +10,303 @@
     });
+    return {
+      success: false,
+      reasons: logMessages,
+    };
+  }
 
-    log.info(`Successfully audited ${baseURL} for ${type} type audit`);
+  const { protocol, domain } = parsedUrl;
+  let sitemapUrls = [];
+  try {
+    const robotsResult = await checkRobotsForSitemap(protocol, domain);
+    if (robotsResult.paths.length) {
+      sitemapUrls = robotsResult.paths;
+    }
+  } catch (error) {
+    logMessages.push({ value: `Error fetching or processing robots.txt: ${error.message}`, error: ERROR_CODES.FETCH_ERROR });
+    // Don't return failure yet, try the fallback URLs
+  }
 
-    return noContent();
-  } catch (e) {
-    return internalServerError('Sitemap audit failed');
+  if (!sitemapUrls.length) {
+    const commonSitemapUrls = [`${protocol}://${domain}/sitemap.xml`, `${protocol}://${domain}/sitemap_index.xml`];
+    sitemapUrls = await checkCommonSitemapUrls(commonSitemapUrls);
+    if (!sitemapUrls.length) {
+      logMessages.push({ value: `No sitemap found in robots.txt or common paths for ${protocol}://${domain}`, error: ERROR_CODES.NO_SITEMAP_IN_ROBOTS });
+      return { success: false, reasons: logMessages };
+    }
+  }
+
+  const inputUrlToggledWww = toggleWWW(inputUrl);
+  const filteredSitemapUrls = sitemapUrls.filter(
+    (path) => path.startsWith(inputUrl) || path.startsWith(inputUrlToggledWww),
+  );
+  const extractedPaths = await getBaseUrlPagesFromSitemaps(inputUrl, filteredSitemapUrls);
+
+  if (Object.entries(extractedPaths).length > 0) {
+    logMessages.push({ value: 'Sitemaps found and validated successfully.' });
+    return {
+      success: true, reasons: logMessages, paths: extractedPaths, url: inputUrl,
+    };
+  } else {
+    logMessages.push({ value: 'No valid paths extracted from sitemaps.', error: ERROR_CODES.NO_PATHS_IN_SITEMAP });
+    return { success: false, reasons: logMessages, url: inputUrl };
   }
 }
+
+/**
+ * Performs an audit for a specified site based on the audit request message.
+ *
+ * @async
+ * @param {string} baseURL - The URL to run the audit against.
+ * @param {Object} context - The lambda context object.
+ * @returns {Promise<{fullAuditRef: string, auditResult: Object}>}
+ */
+export async function sitemapAuditRunner(baseURL, context) {
+  const { log } = context;
+  log.info(`Received sitemap audit request for ${baseURL}`);
+  const startTime = process.hrtime();
+  const auditResult = await findSitemap(baseURL);
+
+  const endTime = process.hrtime(startTime);
+  const elapsedSeconds = endTime[0] + endTime[1] / 1e9;
+  const formattedElapsed = elapsedSeconds.toFixed(2);
+
+  log.info(`Sitemap audit for ${baseURL} completed in ${formattedElapsed} seconds`);
+
+  return {
+    fullAuditRef: baseURL,
+    auditResult,
+    url: baseURL,
+  };
+}
+
+export default new AuditBuilder()
+  .withRunner(sitemapAuditRunner)
+  .withUrlResolver((site) => composeAuditURL(site.getBaseURL())
+    .then((url) => (getUrlWithoutPath(prependSchema(url)))))
```

Comment:
looks like we use this url manipulation logic in many places. We might think of either creating a new helper or extending the `composeAuditURL` function cc/ @dzehnder